crates/cext/src/circuit_library/iqp.rs

```diff
@@ -0,0 +1,109 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2025
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use ndarray::ArrayView2;
+use qiskit_circuit::{circuit_data::CircuitData, operations::Param};
+use qiskit_circuit_library::iqp::{check_symmetric, iqp, py_random_iqp};
+
+/// Generate an IQP circuit from an integer interaction matrix.
+///
+/// The `interactions` matrix is interpreted as an `n x n` row-major array of
+/// 64-bit integers, where `n = num_qubits`. The diagonal entries set T-like
+/// phase powers, and the upper triangle encodes two-qubit CPhase interactions.
+///
+/// # Parameters
```

Comment:
Could you update this to match the C documentation style? We're using doxygen syntax, see e.g. https://github.com/Qiskit/qiskit/blob/d76248551f38ae56ff6352098486aa20f512dad5/crates/cext/src/sparse_observable.rs#L115